samza-azure/src/main/java/org/apache/samza/system/azureblob/avro/AzureBlobAvroWriter.java

```diff
@@ -108,19 +109,32 @@ public class AzureBlobAvroWriter implements AzureBlobWriter {
   private final String blobURLPrefix;
   private final long maxBlobSize;
   private final long maxRecordsPerBlob;
+  private final int initBufferSize;
   private final boolean useRandomStringInBlobName;
   private final Object currentDataFileWriterLock = new Object();
   private volatile long recordsInCurrentBlob = 0;
   private BlobMetadataGeneratorFactory blobMetadataGeneratorFactory;
   private Config blobMetadataGeneratorConfig;
   private String streamName;
 
+  @Deprecated
```

Comment:
Why are you deprecating this? If the user does not care about the size, it can always start with the default size.

Reply:
I was deprecating this because it wasn't clear to me that the class _should_ have multiple constructors. And marking as deprecated avoids the major version bump requirement for public APIs. I believe the expectation is that config values are not expected to be referenced throughout the code, but maybe default values are acceptable. 

I will follow up with Samza committers.

Reply:
Is this constructor used somewhere else other than the factory? If not, I'd go ahead and remove this code since the factory handles the logic on passing in the buffer size.

The contract for this class is clear that going forward buffer size needs to be provided.

Reply:
It isn't used elsewhere in the samza codebase. It has been removed.

Reply:
I though you were going to remove this? Why keep this constructor around?